docs/tpm-simulators.md

```diff
@@ -0,0 +1,170 @@
+these measurements, without them there is nothing to attest to, which makes use
+of simulated TPM useless for practical applications. It also means that nothing
+called `TPM2_Startup` command. Attester is able to detect this and calls that
+function automatically.
+
+Hardware and firmware TPM vendors give assurance of proper functioning of their
+TPMs. This is done by creating a certificate for each TPM separately,
+specifically for their Endorsement Keys (EKs). These certificates are created
+during TPM manufacturing and provisioning process, and are written to NVRAM.
+They point to their signing certificates through `authorityInfoAccess` X509V3
+extension. The chain continues until a self-signed root that is implicitly
+trusted. That root must be known by Fobnail Token in advance in order to trust
+given TPM.
+
+For simulated TPM, such certificate has to be manually created and injected into
+NVRAM. EK certificate is created for EK created from EPS (Endorsement Primary
+Seed), which is randomly created when the simulator is started for the first
+time. Nobody is issuing certificates for all instances of simulated TPMs. One of
+the reasons is probably the fact that emulators are not fully compatible with
+specification, especially when it comes to hardware protection mechanisms
+(Physical Presence, protected storage etc.).
+
+There are also other small differences, e.g. after non-orderly shutdown `safe`
+field in `TPMS_CLOCK_INFO` structure is set after 12 seconds in simulator, while
+specification allows for up to 2^22 milliseconds (around 70 minutes) to help
+with NVRAM wear leveling. Note that this field is checked by Fobnail during
+attestation.
+
+## Provisioning simulated TPM
+
+As mentioned, EK certificate must be written to TPM NVRAM. In addition, it must
+point to valid CA certificate that will be downloaded during platform
+provisioning. A script for simplifying this process and required configuration
+are included in [Attester's repository](https://github.com/fobnail/fobnail-attester/tree/main/tools).
+`tpm2-tools` and `openssl` are used by this script, so they must be installed.
+
+To create root certificate, EK certificate and write the latter to NVRAM, it is
+enough to call the following (`-s` tells to send `TPM2_Startup` command):
+
+```
+$ ./tools/tpm_manufacture.sh -s
+Sending TPM2_Startup command
+Generating a RSA private key
+................................................+++++
+................+++++
+writing new private key to '/home/user/fobnail-attester/tools/keys_and_certs/ca_priv.pem'
+-----
+Signature ok
+subject=C = PL, O = Fobnail, ST = State, CN = EK certificate
+Getting CA Private Key
+
+
+Done.
+To test:
+        tpm2_nvread -C o 0x01C00002 | openssl x509 -text -noout -inform DER
+```
+
+Check if certificate was written properly by executing suggested command:
+
+```
+$ tpm2_nvread -C o 0x01C00002 | openssl x509 -text -noout -inform DER
```

Comment:
EK is correctly returned, but I'm getting this from tpm2_nvread

https://paste.dasharo.com/?a2e8d0159252d22f#FYPHwA2j8BJ4R72DoYAEE5UXfTFEKEfQu2Rgk5ueg8cv

Reply:
Ok, I see following comment.